host_core/lib/host_core.ex

```diff
@@ -1,18 +1,81 @@
 defmodule HostCore do
   @moduledoc """
-  Documentation for `HostCore`.
+  `HostCore` Application.
   """
+  require Logger
+  use Application
 
-  @doc """
-  Hello world.
+  def start(_type, _args) do
+    config = Vapor.load!(HostCore.ConfigPlan)
 
-  ## Examples
+    children = mount_supervisor_tree(config)
 
-      iex> HostCore.hello()
-      :world
+    opts = [strategy: :one_for_one, name: HostCore.Supervisor]
 
-  """
-  def hello do
-    :world
+    Logger.info("Starting Host Core")
+    Supervisor.start_link(children, opts)
+  end
+
+  defp mount_supervisor_tree(config) do
```

Comment:
While I like the bundling of all the supervisor logic in a `mount_supervisor_tree` function, I am not comfortable with moving it out of the `Application` module. This module is expected by Elixir developers to be the entry point into the supervision hierarchy, so by moving that code here, we'd be making it harder for people to find the supervision tree.

Reply:
Hi @autodidaddict Thanks for your review.. 
There is actually no dependency for elixir developers on the module application. I already saw many projects and worked on many projects where the root module was the module that implemented the behavior Application. So it's much more of an individual preference than a mandatory or collective good practice

If this is mandatory for this project I can change it, however I would prefer to keep it as it is.

Reply:
I don't have a strong preference in either direction, priority here would be ease of use for other elixir developers. From the documentation:
```
Applications are the idiomatic way to package software in Erlang/OTP. To get the idea, they are similar to the "library" concept common in other programming languages, but with some additional characteristics.
```

Further down, I can find an example where the root module is using `use Application`
https://hexdocs.pm/elixir/1.12/Application.html#module-the-application-callback-module

One of the biggest elixir projects uses this pattern https://github.com/phoenixframework/phoenix/blob/master/lib/phoenix.ex, and I see it pretty commonly now too. I'm fine with it the way it's proposed in the PR after doing a bit more research

Reply:
Thanks for the comment @brooksmtownsend